core/src/com/jediterm/terminal/model/TerminalTextBuffer.java

```diff
@@ -121,10 +121,18 @@ private LinesBuffer createHistoryBuffer() {
     if (newHeight < oldHeight) {
       if (!myAlternateBuffer) {
         int lineDiffCount = oldHeight - newHeight;
-        //we need to move lines from text buffer to the scroll buffer
-        //but empty bottom lines up to the cursor can be collapsed
+        // We need to move lines from text buffer to the scroll buffer,
+        // but empty bottom lines up to the cursor can be collapsed
+
+        // Number of lines to remove until the new height or cursor if it is located below the new height.
         int maxBottomLinesToRemove = Math.min(lineDiffCount, Math.max(0, oldHeight - oldCursorY));
-        int emptyLinesDeleted = myScreenBuffer.removeBottomEmptyLines(oldHeight - 1, maxBottomLinesToRemove);
+        // Number of already empty lines on the screen (but not greater than required count to remove)
+        int emptyLinesCount = Math.min(maxBottomLinesToRemove, oldHeight - myScreenBuffer.getLineCount());
```

Comment:
Probably, it'd be safer to have `Math.max(0, oldHeight - myScreenBuffer.getLineCount())`.

Also, it'd be a bit better to rearrange commits, so "Use more time effective cyclic buffer for storing terminal lines" uses the updated signature. But both are minor and can be neglected for now.

Reply:
Yes, maybe for general safety we should ensure that it is not negative.
But now `oldHeight` is either the real old height or the height after width resize. In both cases it is guaranteed that buffer size is not grater.